src/parallel/job_token.rs

```diff
@@ -233,19 +233,16 @@ mod inprocess_jobserver {
     impl JobServer {
         pub(super) fn new() -> Self {
             // Use `NUM_JOBS` if set (it's configured by Cargo) and otherwise
-            // just fall back to a semi-reasonable number.
-            //
-            // Note that we could use `num_cpus` here but it's an extra
-            // dependency that will almost never be used, so
-            // it's generally not too worth it.
-            let mut parallelism = 4;
-            // TODO: Use std::thread::available_parallelism as an upper bound
-            // when MSRV is bumped.
-            if let Ok(amt) = var("NUM_JOBS") {
-                if let Ok(amt) = amt.parse() {
-                    parallelism = amt;
-                }
-            }
+            // just fall back to the number of cores on the local machine.
+
+            let parallelism =
+                if let Some(num_jobs) = var("NUM_JOBS").ok().and_then(|j| j.parse::<u32>().ok()) {
```

Comment:
Nit: Do something like:
```suggestion
            let num_jobs = var("NUM_JOBS").ok().and_then(|j| j.parse::<u32>().ok())
            let parallelism = if let Some(num_jobs) = num_jobs {
```

(Formats better).

Reply:
(At least I just really dislike the extra indentation the entire `if` block gets.

Reply:
Yeah I also wasn't a fan of that indentation :grin:

Reply:
IMO maybe better to use chained or_else/* calls